src/nvim/insexpand.c

```diff
@@ -2929,10 +2935,19 @@ static void get_complete_info(list_T *what_list, dict_T *retdict)
     }
     if (ret == OK && (what_flag & CI_WHAT_SELECTED)) {
       ret = tv_dict_add_nr(retdict, S_LEN("selected"), selected_idx);
-      win_T *wp = win_float_find_preview();
-      if (wp != NULL) {
-        tv_dict_add_nr(retdict, S_LEN("preview_winid"), wp->handle);
-        tv_dict_add_nr(retdict, S_LEN("preview_bufnr"), wp->w_buffer->handle);
+    }
+  }
+
+  // Support for preview window and preview buffer information
+  if (ret == OK && (what_flag & (CI_WHAT_PREVIEWWIN | CI_WHAT_PREVIEWBUF))) {
+    win_T *wp = win_float_find_preview();
+    if (wp != NULL) {
+      if (what_flag & CI_WHAT_PREVIEWWIN) {
+        ret = tv_dict_add_nr(retdict, S_LEN("preview_winid"), wp->handle);
+      }
+
+      if (ret == OK && (what_flag & CI_WHAT_PREVIEWBUF)) {
+        ret = tv_dict_add_nr(retdict, S_LEN("preview_bufnr"), wp->w_buffer->handle);
```

Comment:
Why do we need both of these fields? The buffer shown by `preview_winid` is implicitly the "preview bufnr".

Reply:
yes .. It is just a convenience. users do not need nvim_win_get_buf or winbufnr

Reply:
Let's drop it for now. That is not a very strong use-case for increasing surface area here.

Keeping the surface-area minimal will help us focus on the important parts of this interface (which we are still figuring out).

Reply:
if we want drop the `nvim__complete_set` also need remove bufnr from result dict right ?